tests/core/test_storage.py

```diff
@@ -342,6 +342,17 @@ def test_delete_raise_when_unknown(self):
             **self.storage_kw
         )
 
+    def test_get_all_handles_parent_id_pattern_matching(self):
+        self.create_record(parent_id='abc', collection_id='c')
+        record = self.create_record(parent_id='abc', collection_id='c')
+        self.storage.delete(object_id=record['id'], parent_id='abc', collection_id='c')
+        self.create_record(parent_id='efg', collection_id='c')
+
+        records, total_records = self.storage.get_all(parent_id='ab*', collection_id='c',
+                                                      include_deleted=True)
+        self.assertEquals(len(records), 2)
+        self.assertEquals(total_records, 1)
+
```

Comment:
I would suggest to add a test without parent matching to make sure that we kept the strict comparison behaviour when two parent_ids are similar (eg. `parent_id: xabcx` should not be returned with `parent_id=abc`)

By the way, maybe we should have it for `delete_all()` also...